### PR TITLE
Added driver support for m24256 eeprom

### PR DESCRIPTION
**Pull request by ghost**
Added the driver support for m24256 eeprom. It allows to read and write the data

### CODE REVIEW

**Comment by Arunmanialagarsamy** [comment]
@henrikbrixandersen Please take a review

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen Please take a review

I will get to it, you only submitted this 4 days ago. While you wait, I suggest you read up on our documented Contributor Expectations, especially the section on review escalation: https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#pr-review-escalation

**Comment by Arunmanialagarsamy** [comment]
@henrikbrixandersen Waiting for your approval, please take a look.

File: samples/drivers/eeprom/boards/b_u585i_iot02a.overlay
Comment (kartben) [inline samples/drivers/eeprom/boards/b_u585i_iot02a.overlay:12]: The datasheet for this board suggests it's 256Kbits?

File: samples/drivers/eeprom/boards/b_u585i_iot02a.overlay
Comment (kartben) [inline samples/drivers/eeprom/boards/b_u585i_iot02a.overlay:1]: Don't make this an overlay and directly add to the board's DTS instead. You may also update the. Boards readme accordingly.

File: dts/bindings/mtd/st,m24256.yaml
Comment (kartben) [inline dts/bindings/mtd/st,m24256.yaml:7]: I am guessing this could be a generic st,m24xxx driver, no? i.e. would a m24512 driver really be different?
Interestingly, `boards/shields/x_nucleo_eeprma2` already uses an `st,m24xxx` compatible :)

**Review by kartben — CHANGES_REQUESTED** [review]

File: dts/bindings/mtd/st,m24256.yaml
Comment (henrikbrixandersen) [reply dts/bindings/mtd/st,m24256.yaml:7]: Perhaps this can be made even more generic. From a quick glance, it looks very similar to https://github.com/zephyrproject-rtos/zephyr/pull/69327

Comment (benediktibk) [reply dts/bindings/mtd/st,m24256.yaml:7]: Just as a hint, M24M02 seems to be different though:
image

It uses part of the I2C address for the memory address and therefore one I2C slave uses up multiple I2C addresses on the bus.

So probably we shouldn't be too catch-all with the compatible string.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
There are so many similarities with the driver submitted in #69327 that I think these can be combined into one driver, supporting this "class" of I2C EEPROMs. See example in `drivers/eeprom/eeprom_at2x.c` which supports both Atmel AT24 and AT25 series (and compatibles) with one driver.

Please coordinate internally between the two submitters on how to proceed with one, common driver. I'll gladly participate in a discussion about this, either on the Zephyr Discord server, in an Architecture WG meeting or elsewhere.

**Review by str4t0m — CHANGES_REQUESTED** [review]
This eeprom should already work with the `atmel,at24` driver as is.
Could you please give it a try on your b_u585i_iot02a board.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
The st,m24xxx EEPROM family is already supported by the existing `eeprom_at2x.c` driver. Please see the following shield for how to describe these via devicetree:

https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/shields/x_nucleo_eeprma2/x_nucleo_eeprma2.overlay

**Comment by Arunmanialagarsamy** [comment]
Lets close this PR as per our discussion
https://discordapp.com/channels/720317445772017664/1218200905044721765/1219957042874155008

**Pull request closed without merging**
